### ISSUE TITLE
help ERROR:  pllua: cannot call into PostgreSQL with pending errors

### ISSUE THREAD

**Issue by exqmjmz** [pllua/pllua#15]
i need help. this is my pl/lua code.can be created successfully.
but after i executed the function
an error occurred

```
cannot call into PostgreSQL with pending errors
```
What caused this?
How can I solve it?
If this error occurs again in the future, how should I troubleshoot it?

```
DROP FUNCTION IF EXISTS get_data_by_page(tablename VARCHAR);

CREATE OR REPLACE FUNCTION get_data_by_page(tablename VARCHAR)
  RETURNS TABLE (
		id uuid,
		createduser varchar,
    createdtime bigint,
		updateduser varchar,
		updatedtime BIGINT,
    data jsonb
  ) LANGUAGE pllua stable AS $$
		print(tablename)
  	local r = spi.rows('select id,createduser,createdtime,updateduser,updatedtime,data from $1',tablename)
		for k in r do 
				coroutine.yield(k)
		end
$$;

SELECT * from get_data_by_page('client_info');
```

**Comment by RhodiumToad** [comment]
It looks like this is happening when a PostgreSQL error is raised within a RETURNS TABLE or RETURNS SETOF function, because of an oversight in the error handling path (the code that tries to handle the error ends up calling back into pg too soon). The real error is in your query text - you can't use $1 to replace a table name.

I need to find a way to fix the error handling so that the real error is shown, that may take me a little work.

You should find that the correct error is shown if run in a DO-block or a non-SETOF non-TABLE function.

**Comment by RhodiumToad** [comment]
Hm, I didn't expect that commit to close this. Regardless, I think I have fixed it, and the fixed version is up in the other repo (as linked above) if you want to test it. The fix will be released as 2.0.13 once I've done my own testing.
